scripts/perftune.py

```diff
@@ -332,9 +333,20 @@ def auto_detect_irq_mask(cpu_mask, cores_per_irq_core):
 
         return run_hwloc_calc(['--restrict', cpu_mask] + hwloc_args)
 
+# @param headers dict of k:v
+def curl(url, headers=None, method=None, byte=False, timeout=0):
+    req = urllib.request.Request(url, headers=headers or {}, method=method)
+    with urllib.request.urlopen(req, timeout=timeout) as res:
+        if byte:
+            return res.read()
+        else:
+            return res.read().decode('utf-8')
```

Comment:
Seems like "curl" is adding very little. You can just call urrlib.request.urlopen() in the caller, no? Just add .decode() when byte=False.

Reply:
Fixed on latest push